crates/vm/src/arch/integration_api.rs

```diff
@@ -201,7 +201,7 @@ where
         Self {
             adapter,
             core,
-            records: vec![],
+            records: Vec::with_capacity(1 << 20),
```

Comment:
this should use `access_capacity`

Reply:
Do you have any suggestions for how to access `config` from here?

(Also I think this is a different parameter than `access_capacity`; `access_capacity` is the number of record IDs, which should be several factors larger than any one chip's `records.len()`, both because every chip contributes to memory records and because each chip record corresponds to several memory records.)

Reply:
I guess we could add it as a field in `OfflineMemory`...

Reply:
It seems weird because this is more about the number of instructions executed by any one chip rather than anything memory-related.

Reply:
ok we can address later then

Reply:
at least make it a private constant